src/pages/settings/Profile/PersonalDetails/AddressPage.js

```diff
@@ -52,179 +52,159 @@ const defaultProps = {
-            errors.zipPostCode = this.props.translate('privatePersonalDetails.error.incorrectZipFormat');
         }
-
-        return errors;
+    } else if (!CONST.GENERIC_ZIP_CODE_REGEX.test(values.zipPostCode.trim())) {
+        errors.zipPostCode = translate('privatePersonalDetails.error.incorrectZipFormat');
     }
 
-    render() {
-        const address = lodashGet(this.props.privatePersonalDetails, 'address') || {};
-        const [street1, street2] = (address.street || '').split('\n');
-
-        return (
-            <ScreenWrapper includeSafeAreaPaddingBottom={false}>
-                <HeaderWithCloseButton
-                    title={this.props.translate('privatePersonalDetails.homeAddress')}
-                    shouldShowBackButton
-                    onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS_PERSONAL_DETAILS)}
-                    onCloseButtonPress={() => Navigation.dismissModal(true)}
-                />
-                <Form
-                    style={[styles.flexGrow1, styles.ph5]}
-                    formID={ONYXKEYS.FORMS.HOME_ADDRESS_FORM}
-                    validate={this.validate}
-                    onSubmit={this.updateAddress}
-                    submitButtonText={this.props.translate('common.save')}
-                    enabledWhenOffline
-                >
-                    <View style={styles.mb4}>
-                        <AddressSearch
-                            inputID="addressLine1"
-                            label={this.props.translate('common.addressLine', {lineNumber: 1})}
-                            defaultValue={street1 || ''}
-                            isLimitedToUSA={false}
-                            renamedInputKeys={{
-                                street: 'addressLine1',
-                                street2: 'addressLine2',
-                                city: 'city',
-                                state: 'state',
-                                zipCode: 'zipPostCode',
-                                country: 'country',
-                            }}
-                            maxInputLength={CONST.FORM_CHARACTER_LIMIT}
-                        />
-                    </View>
-                    <View style={styles.mb4}>
-                        <TextInput
-                            inputID="addressLine2"
-                            label={this.props.translate('common.addressLine', {lineNumber: 2})}
-                            defaultValue={street2 || ''}
-                            maxLength={CONST.FORM_CHARACTER_LIMIT}
+    return errors;
+};
+
+/**
+ * Submit form to update user's first and last legal name
+ * @param {Object} values - form input values
+ */
+const updateAddress = (values) => {
+    PersonalDetails.updateAddress(values.addressLine1.trim(), values.addressLine2.trim(), values.city.trim(), values.state.trim(), values.zipPostCode.trim(), values.country);
+};
+
+const AddressPage = (props) => {
```

Comment:
I just read the guidelines and it seems like we prefer declaring components with the [function keyword](https://github.com/Expensify/App/blob/4b1c3264cb0d94c7eba5c89b86003934b0ecd3ac/contributingGuides/STYLE.md#should-i-declare-my-components-with-arrow-functions-const-or-the-function-keyword) instead, which I totally missed in my PRs 😅

Reply:
Ty, I haven't really seen the guidelines, I'll read them now :)

Reply:
Haha all good. I totally missed them until now and already pushed some PRs using `const`

Reply:
Do you know if this applies also to functions outside the component?

Reply:
Not sure. Historically, we've been using the `function` keyword. However, we seem to be using them inconsistently so 🤷

Reply:
ok, I'll just use `function`.